shared/reducers/fs.tsx

```diff
@@ -326,28 +353,35 @@ export default Container.makeReducer<FsGen.Actions, Types.State>(initialState, {
       newFolderName = `New Folder ${i}`
     }
 
-    draftState.edits = draftState.edits.set(
-      Constants.makeEditID(),
-      Constants.makeNewFolder({
-        hint: newFolderName,
-        name: newFolderName,
-        parentPath,
-      })
-    )
+    draftState.edits = new Map([
+      ...draftState.edits,
+      [
+        Constants.makeEditID(),
+        {
+          ...Constants.emptyNewFolder,
+          hint: newFolderName,
+          name: newFolderName,
+          parentPath,
+        },
+      ],
+    ])
   },
   [FsGen.newFolderName]: (draftState, action) => {
-    draftState.edits = draftState.edits.update(
-      action.payload.editID,
-      edit => edit && edit.set('name', action.payload.name)
-    )
+    updateExistingEdit(draftState, action.payload.editID, draftEdit => {
+      draftEdit.name = action.payload.name
+    })
   },
   [FsGen.commitEdit]: (draftState, action) => {
-    draftState.edits = draftState.edits.update(action.payload.editID, edit =>
-      edit.set('status', Types.EditStatusType.Saving)
-    )
+    updateExistingEdit(draftState, action.payload.editID, draftEdit => {
+      draftEdit.status = Types.EditStatusType.Saving
+    })
   },
   [FsGen.discardEdit]: (draftState, action) => {
-    draftState.edits = draftState.edits.remove(action.payload.editID)
+    if (draftState.edits.has(action.payload.editID)) {
+      const edits = new Map([...draftState.edits])
```

Comment:
why not `const edits = new Map(draftState.edits)`? cause `new Map(undefined)`? if so maybe
`new Map(draftState.edits || [])`

Reply:
(and below)

Reply:
Nope; I just didn't know! Will cahnge